api/src/main/java/org/pmiops/workbench/cdr/dao/CustomCBCriteriaDaoImpl.java

```diff
@@ -59,21 +61,50 @@ public MapSqlParameterSource getParameters() {
           + "and (%s)\n"
           + "order by est_count desc, name asc\n";
 
+  private static final String AUTO_COMPLETE_ENDS_WITH_WITHOUT_TERM =
+      "select *\n"
+          + "from %s.cb_criteria\n"
+          + "where type = :type\n"
+          + "and is_standard = :standard\n"
+          + "and has_hierarchy = 1\n"
+          + "and match(full_text) against(concat('+[', :domain, '_rank1]') in boolean mode)\n"
+          + "and (%s)\n"
+          + "order by est_count desc, name asc\n";
+
+  private static final String AUTO_COMPLETE_ENDS_WITH_WITH_TERM =
+      "select *\n"
+          + "from %s.cb_criteria\n"
+          + "where type = :type\n"
+          + "and is_standard = :standard\n"
+          + "and has_hierarchy = 1\n"
+          + "and match(full_text) against(concat(:term, '+[', :domain, '_rank1]') in boolean mode)\n"
+          + "and (%s)\n"
+          + "order by est_count desc, name asc\n";
+
   private static final String DYNAMIC_SQL = "upper(name) like upper(%s)";
 
   private static final String LIMIT_OFFSET = "limit %s offset %s\n";
 
   private static final String OR = "\nor\n";
 
+  private static final String BIND_VAR_DOMAIN = "domain";
+  private static final String BIND_VAR_STANDARD = "standard";
+  private static final String BIND_VAR_TERM = "term";
+  private static final String BIND_VAR_TYPE = "type";
```

Comment:
I would move these to the very top of the class, so they can be used in the static sql statements as well.

Reply:
done, also pulled other SQL snip statics up.